app/scripts/modules/core/src/pipeline/executions/execution/ExecutionMarker.tsx

```diff
@@ -81,15 +84,65 @@ export class ExecutionMarker extends React.Component<IExecutionMarkerProps, IExe
     });
   };
 
+  private manualJudgmentStatus = (stageStatus: string, manualJudgment: any) => {
+    let status = stageStatus;
+    if (manualJudgment !== undefined && manualJudgment.length && stageStatus === 'running') {
+      const existStages = this.props.stage.stages.filter((stage) => stage.status.toLowerCase() === 'running');
+      existStages.forEach(({ context, type }) => {
+        if (this.leafNodeExist(context.executionId, manualJudgment) && type === 'pipeline') {
+          status = 'waiting';
+        }
+      });
+    }
+    return status;
+  };
+
+  private leafNodeExist = (executionContextId: string, manualJudgment: any): string => {
+    const nestedLeafnodeObj = this.props.manualJudgment[executionContextId];
+    const leafNodeObj = manualJudgment.find(
+      (leafnode: any) => leafnode.currentChild === executionContextId || leafnode.id === executionContextId,
+    );
+    if (!nestedLeafnodeObj && !leafNodeObj) return null;
+    if (nestedLeafnodeObj) {
+      return this.leafNodeExist(nestedLeafnodeObj[0].pipelineId ?? nestedLeafnodeObj[0].id, nestedLeafnodeObj);
+    } else if (leafNodeObj) {
+      return leafNodeObj;
+    }
+    return null;
+  };
+
+  private redirectLeafNode = (type: string, manualJudgment: any, contextExecutionId: string): any => {
+    if (manualJudgment !== undefined) {
+      const leafnode = this.leafNodeExist(contextExecutionId, manualJudgment);
+      if (leafnode) {
+        return this.fetchLeafNodeParameter(type, leafnode);
+      }
+    }
+  };
+
+  private fetchLeafNodeParameter = (type: string, leafNodeObject: any) => {
+    let leafNodeVal = '';
+    if (!isEmpty(leafNodeObject)) {
+      if (type === 'application') {
+        leafNodeVal = leafNodeObject.app ?? this.props.application.name;
+      } else {
+        leafNodeVal = leafNodeObject.id;
+      }
+    }
+    return leafNodeVal;
+  };
+
   public render() {
-    const { stage, application, execution, active, previousStageActive, width } = this.props;
+    const { stage, application, execution, active, previousStageActive, width, manualJudgment } = this.props;
     const stageType = (stage.activeStageType || stage.type).toLowerCase(); // support groups
+    const PIPELINE_WAITING =
+      this.manualJudgmentStatus(stage.status.toLowerCase(), manualJudgment[execution.id]) === 'waiting';
```

Comment:
Since you call this function a couple times, lets extract into a const.
```
const manualJudgmentStatus = this.manualJudgmentStatus(stage.status.toLowerCase(), manualJudgment[execution.id])
```